pkg/config/environment_detection.go

```diff
@@ -116,13 +118,39 @@ func detectContainerFeatures() {
 	}
 }
 
+func checkSocketExists(path string) bool {
+	f, err := os.Stat(path)
+	if err != nil {
+		return false
+	}
+
+	// On Windows, we cannot easily verify that a path is a named pipe
+	if runtime.GOOS == "windows" {
+		return true
+	}
+
+	if !f.IsDir() && f.Mode()&os.ModeSocket > 0 {
```

Comment:
Few nits:
- I would think that `f.Mode()&os.ModeSocket` would be correct without the dir check in the front. If you leave things as-is, some parens would be nice around the flag logic to make it clear about the order of eval.
- I think generally I've seen flag selection done with `!= 0` instead of `>0` but definitely doesn't matter much